diff --git a/src/injected.ts b/src/injected.ts
index 688d9f947..fb2abc430 100644
--- a/src/injected.ts
+++ b/src/injected.ts
@@ -2,6 +2,36 @@
 
 console.log('[Rocket.Chat Desktop] Injected.ts');
 
+console.log('[Rocket.Chat Desktop] Injecting Ubuntu Touch keyboard responsiveness');
+
+var lastClickEl;
+window.addEventListener("click", function() {
+  //Register Last clicked element
+  lastClickEl=event.target;
+  console.log(lastClickEl);
+});
+
+document.body.addEventListener('focusin', (event) => {
+    setTimeout(function () {
+      if ( event.target === document.querySelector('[name="msg"]') && document.querySelector('[name="msg"]').parentElement.parentElement.contains(lastClickEl) )
+      {
+      var keyboardMargin = 12; //px
+      var keyboardHeight = Math.round(parseFloat(window.__cmdParams.keyboardHeight) /parseFloat(window.__cmdParams.forceScale)+keyboardMargin);
+      document.querySelector('[name="msg"]').parentElement.parentElement.style.paddingBottom=`${keyboardHeight}px`;
+      }
+      else
+      {
+          console.log(lastClickEl);
+          document.querySelector('[name="msg"]').parentElement.parentElement.style.paddingBottom="0";
+          if ( event.target === document.querySelector('[name="msg"]') )
+            event.target.blur();
+      }
+      }, 300);
+});
+
+
+console.log('[Rocket.Chat Desktop] End of injecting Ubuntu Touch keyboard responsiveness');
+
 const resolveWithExponentialBackoff = <T>(
   fn: () => Promise<T>,
   { maxRetries = 5, delay = 1000 } = {}

diff --git a/src/ui/main/serverView/index.ts b/src/ui/main/serverView/index.ts
index db7c005b9..500dc9e66 100644
--- a/src/ui/main/serverView/index.ts
+++ b/src/ui/main/serverView/index.ts
@@ -586,17 +586,33 @@ export const attachGuestWebContentsEvents = async (): Promise<void> => {
       )?.[0]
   );
 
+
+
+  let rawInjectedCode: string | undefined;
   let injectableCode: string | undefined;
   handle('server-view/ready', async (webContents) => {
-    if (!injectableCode) {
-      injectableCode = await fs.promises.readFile(
-        path.join(app.getAppPath(), 'app/injected.js'),
-        'utf8'
-      );
+    if (!rawInjectedCode) {
+      rawInjectedCode = await fs.promises.readFile(
+          path.join(app.getAppPath(), 'app/injected.js'),
+          'utf8'
+        );
     }
 
+  const params = {
+        forceScale: app.commandLine.getSwitchValue('force-device-scale-factor'),
+        keyboardHeight: app.commandLine.getSwitchValue('keyboard-height'),
+  };
+
+
+  injectableCode = `
+  window.__cmdParams = ${JSON.stringify(params)};
+
+  ${rawInjectedCode}
+  `;
+
     webContents.executeJavaScript(injectableCode, true);
 
+
     if (process.env.NODE_ENV === 'development') {
       injectableCode = undefined;
     }
